### PR TITLE
🚧 New way for adding navigation bar elements

### LINKED ISSUE #2: Fix Navigation Bar Elements ignoring State variables.
- [x] I have checked that nobody else has reported this issue.
- [x] I know that I'm reporting a bug for Codename Starlight (SwiftUI) and not for Hyperspace Desktop (JS).

**Describe the bug**

When creating a navigation bar element, there are many times where you want it to change depending on a specific variable. The problem with this is that, since navigation bar elements are passed through preference keys, for some reason they seem to ignore state variable updates, and, as a result, do not change.

### PR DESCRIPTION
**Pull request by alexmodrono**
**Changes Overview**  
This PR makes the following changes:

- **BREAKING CHANGE**: `.navigationBarElement()` no longer works (the function is still there though it will probably be deleted before the final version).
- Introduces a new way for adding navigation bar elements that replaces the old way.
- Allows for more customization of elements thanks to `NavigationBarProxy`.

**Does this PR fix, close, or implement any issues?**  
- [x] This PR closes, fixes, or implements the following issues.

- Fixes #2 

**Pending for review**  
@hyperspacedev/swiftui

### CODE REVIEW

File: Sources/StylableScrollView/Extensions/Modifiers.swift [hunk 1]
```diff
@@ -85,18 +86,21 @@ public extension ScrollView {
...
     ///
-    @ViewBuilder @inlinable func stretchableHeader<Title, NavBar>(
+    @ViewBuilder @inlinable func stretchableHeader<Title, NavBar, TrailingElements, LeadingElements>(
         header: Image,
         title: () -> Title,
-        navBar: () -> NavBar
-    ) -> some View where Title: View, NavBar: View {
+        navBar: () -> NavBar,
+        trailingElements: @escaping (NavigationBarProxy) -> TrailingElements,
+        leadingElements: @escaping (NavigationBarProxy) -> LeadingElements
```
Comment (alicerunsonfedora) [inline Sources/StylableScrollView/Extensions/Modifiers.swift]: Shouldn't this be the other way around, from leading to trailing?

**Review by alicerunsonfedora — DISMISSED** [review]
Overall looks good, but I'd consider swapping trailing and leading elements in the corresponding signatures.

**Review by alicerunsonfedora — APPROVED** [review]

File: Sources/StylableScrollView/Extensions/Modifiers.swift (hunk 1 above)
Comment (alexmodrono) [reply Sources/StylableScrollView/Extensions/Modifiers.swift]: Fixed by f36a2b3.

**Comment by alexmodrono** [comment]
Alright, I'll merge this then.

**Pull request merged**